Framework/Algorithms/src/ReflectometryReductionOne.cpp

```diff
@@ -578,29 +590,30 @@ void ReflectometryReductionOne::exec() {
     auto detectorWS = inLam.get<0>();
     auto monitorWS = inLam.get<1>();
 
-    if (isMultiDetector) {
-      if (directBeam.is_initialized()) {
-        // Sum over the direct beam.
-        WorkspaceIndexList db = directBeam.get();
-        std::stringstream buffer;
-        buffer << db.front() << "-" << db.back();
-        MatrixWorkspace_sptr regionOfDirectBeamWS = this->toLamDetector(
-            buffer.str(), runWS, wavelengthInterval, wavelengthStep);
+    if (detectorEfficiencyCorrection.is_initialized()) {
+      MatrixWorkspace_sptr decWS = *detectorEfficiencyCorrection;
+      if (decWS->blocksize() > 1) {
+        // If using separate normalization constants for each spectra, rebin
+        // to the detector workspace
+        if (decWS->blocksize() < detectorWS->blocksize()) {
+          throw std::invalid_argument("The number of y-values in the detector "
```

Comment:
There's something here I don't understand. According to the issue, there are two possibilities for the `DetectorEfficiencyCorrection` workspace:

1. It is a workspace with a single spectrum and a single value in this spectrum
2. It is a workspace with `N` spectra and a single value per spectrum.

@PranavBahuguna Am I correct about this? If so, there's no point in checking `blocksize()` as it'll always be 1. In fact, the condition:

```
decWS->blocksize() < detectorWS->blocksize()
```

should/must always be true. What we should be checking is that `blocksize() == 1` and that if the number of spectra in `decWS` (if more than one) matches the number of spectra in `detectorWS`.